sdk/src/timing.rs

```diff
@@ -81,19 +81,19 @@ impl AtomicInterval {
     /// except, if skip_first=false, false until 'interval_time_ms' has elapsed since this struct was created
     pub fn should_update_ext(&self, interval_time_ms: u64, skip_first: bool) -> bool {
         let now = timestamp();
-        let last = self.last_update.load(Ordering::Relaxed);
+        let last = self.last_update.load(Ordering::Acquire);
         now.saturating_sub(last) > interval_time_ms
             && self
                 .last_update
-                .compare_exchange(last, now, Ordering::Relaxed, Ordering::Relaxed)
+                .compare_exchange(last, now, Ordering::AcqRel, Ordering::Acquire)
```

Comment:
I think the second ordering can stay `Relaxed` since the failure value is not used.